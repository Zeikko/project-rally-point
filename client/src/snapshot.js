import renderer from 'react-test-renderer'
import 'jest-styled-components'

export function matchComponentToSnapshot(component) {
  const json = renderer.create(component).toJSON()
  expect(json).toMatchSnapshot()
}