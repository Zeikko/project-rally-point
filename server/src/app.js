import express from 'express'
import path from 'path'
import session from 'express-session'
import connectPgSimple from 'connect-pg-simple'
import api from './routes/api'
import passport from './passport'
import config from './config'

const PgSession = connectPgSimple(session)
const app = express()

app.use(session({
  secret: 'secret',
  store: new PgSession({
    conString: config.dbUrl,
    pruneSessionInterval: config.pruneSessionInterval,
  }),
  resave: false,
  saveUninitialized: true,
}))
app.use(passport.initialize())
app.use(passport.session())

app.use('/api', api)
app.get('/', (req, res) => res.sendFile('client/html/index.html', { root: path.resolve(__dirname, '../../') }))
app.get('/client.js', (req, res) => res.sendFile('client/dist/index.js', { root: path.resolve(__dirname, '../../') }))
app.use('/img', express.static(path.join(__dirname, '../../client/img')))

export default app
