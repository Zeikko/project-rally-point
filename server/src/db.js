import knex from 'knex'
import config from './config'

export default knex({
  client: 'postgres',
  connection: config.dbUrl,
  useNullAsDefault: true,
})
