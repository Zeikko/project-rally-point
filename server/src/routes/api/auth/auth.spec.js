import supertest from 'supertest'
import passportStub from 'passport-stub'
import { matchBodyToSnapshot } from '../../../test/snapshot'
import app from '../../../app'
import db from '../../../db'
import { migrateLatest, migrateRollback } from '../../../test/db'
import normalUserFixture from '../../../../../fixtures/normal-user.json'

passportStub.install(app)
const request = supertest(app)

describe('/api/auth', () => {
  beforeAll(() => migrateLatest()
    .then(() => db('user').insert([normalUserFixture])))

  afterAll(() => migrateRollback())

  afterEach((done) => {
    passportStub.logout()
    done()
  })

  describe('/login GET', () => {
    it('redirect user', () => request
      .get('/api/auth/login')
      .expect(302))
  })

  describe('/logout POST', () => {
    it('does nothing for non logged in user', () => request
      .post('/api/auth/logout')
      .expect(200)
      .then(matchBodyToSnapshot))

    it('logs user out', () => request
      .post('/api/auth/logout')
      .expect(200)
      .then(matchBodyToSnapshot))
  })

  describe('/loggedin GET', () => {
    it('returns null when not logged in', () => request
      .get('/api/auth/loggedin')
      .expect(200)
      .then(response => expect(response.body).toBeNull()))

    it('returns user when logged in', () => {
      passportStub.login(normalUserFixture.id)
      return request
        .get('/api/auth/loggedin')
        .expect(200)
        .then(response => expect(response.body).toEqual(normalUserFixture))
    })
  })
})

