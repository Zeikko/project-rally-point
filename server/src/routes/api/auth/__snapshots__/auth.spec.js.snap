// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/auth /logout POST does nothing for non logged in user 1`] = `
Object {
  "message": "Logged out",
}
`;

exports[`/api/auth /logout POST logs user out 1`] = `
Object {
  "message": "Logged out",
}
`;
